src/components/buttons/z-button/index.tsx

```diff
@@ -21,30 +21,36 @@ export class ZButton {
   /** name */
   @Prop() name?: string;
   /** disable button */
-  @Prop() disabled?: boolean = false;
+  @Prop({ reflect: true }) disabled?: boolean = false;
   /** button type */
   @Prop() type?: HTMLButtonElement["type"] = ButtonTypeEnum.button;
   /** graphic variant */
-  @Prop() variant?: ButtonVariantBean = ButtonVariantEnum.primary;
+  @Prop({ reflect: true }) variant?: ButtonVariantBean = ButtonVariantEnum.primary;
   /** reduce button size (optional) */
-  @Prop() issmall?: boolean = false;
+  @Prop({ reflect: true }) issmall?: boolean = false;
   /** add an icon to button (optional) */
   @Prop() icon?: string;
 
-  render() {
-    this.hostElement.style.pointerEvents = this.disabled ? "none" : "auto";
-
+  renderLegacyButton() {
     return (
       <button
         id={this.htmlid}
         name={this.name}
         type={this.type}
         disabled={this.disabled}
-        class={`${this.variant} ${this.issmall && "small"}`}
+        class={`${this.variant}${this.issmall ? " small" : ""}`}
       >
         {this.icon && <z-icon name={this.icon} width={16} height={16} />}
         <slot />
       </button>
     );
   }
+
+  render() {
+    return (
+      <slot name="element">
+        {this.renderLegacyButton()}
```

Comment:
Non capisco il naming. Also, perché incapsularlo in un metodo a parte?

Reply:
Ci sembrava più chiaro tenerlo a parte per esplicitare che sia la retrocompatibilità.

Reply:
Quindi ci aspettiamo che qualcuno possa richiamare esplicitamente il metodo `renderLegacyButton`? Come avete in mente un suo utilizzo?

Reply:
No, il metodo è stato creato solo per questioni di manutenibilità del codice. L'ho reso privato per essere più chiaro che non è un metodo da chiamare da fuori.

Reply:
Sono stato volutamente ossimorico perché imho in questo caso non aggiunge nulla n'é in termini di leggibilitá n'é di manutenibilitá. In generale non son un fan dello split in funzioni laddove non serva.

Non é questo il caso, ma piuttosto che fare su e giú per navigare il flow del componente tra tutti i suoi `renderThis` e `renderThat` incoraggerei lo split in componenti piú piccoli.